python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -307,5 +346,100 @@ def predict_proba(self, dataset, iou_threshold=0.1):
+
+
+class PytorchFasterRCNNWrapper(
+        od_common.GeneralObjectDetectionModelWrapper):
+    """Wraps a PytorchFasterRCNN model with a predict API function.
+
+    To be compatible with the D-RISE explainability method,
+    all models must be wrapped to have the same output and input class and a
+    predict function for object detection. This wrapper is customized for the
+    FasterRCNN model from Pytorch, and can also be used with the RetinaNet or
+    any other models with the same output class.
+
+    :param model: Object detection model
+    :type model: PytorchFasterRCNN model
+    :param number_of_classes: Number of classes the model is predicting
+    :type number_of_classes: int
+    """
+
+    def __init__(self, model, number_of_classes: int):
+        """Initialize the PytorchFasterRCNNWrapper."""
+        self._model = model
+        self._number_of_classes = number_of_classes
+
+    def predict(self, x: torch.Tensor):
+        """Create a list of detection records from the image predictions.
+
+        :param x: Tensor of the image
+        :type x: torch.Tensor
+        :return: Baseline detections to get saliency maps for
+        :rtype: List of Detection Records
+        """
+        raw_detections = self._model(x)
+
+        def apply_nms(orig_prediction: dict, iou_thresh: float = 0.5):
+            """Perform nms on the predictions based on their IoU.
+
+            :param orig_prediction: Original model prediction
+            :type orig_prediction: dict
+            :param iou_thresh: iou_threshold for nms
+            :type iou_thresh: float
+            :return: Model prediction after nms is applied
+            :rtype: dict
+            """
+            keep = torchvision.ops.nms(orig_prediction['boxes'],
+                                       orig_prediction['scores'], iou_thresh)
+
+            nms_prediction = orig_prediction
+            nms_prediction['boxes'] = nms_prediction['boxes'][keep]
+            nms_prediction['scores'] = nms_prediction['scores'][keep]
+            nms_prediction['labels'] = nms_prediction['labels'][keep]
+            return nms_prediction
+
+        def filter_score(orig_prediction: dict, score_thresh: float = 0.5):
+            """Filter out predictions with confidence scores < score_thresh.
+
+            :param orig_prediction: Original model prediction
+            :type orig_prediction: dict
+            :param score_thresh: Score threshold to filter by
+            :type score_thresh: float
+            :return: Model predictions filtered out by score_thresh
+            :rtype: dict
+            """
+            keep = orig_prediction['scores'] > score_thresh
+
+            filter_prediction = orig_prediction
+            filter_prediction['boxes'] = filter_prediction['boxes'][keep]
+            filter_prediction['scores'] = filter_prediction['scores'][keep]
+            filter_prediction['labels'] = filter_prediction['labels'][keep]
+            return filter_prediction
```

Comment:
If both `apply_nms` and `filter_score` have the same function definitions in the 2 wrappers, consider having an interface wrapper that defines these functions, and have the two wrappers inherit the functions off of that interface and define their own `predict` & `predict_proba`.